app/src/main/java/org/gdg/frisbee/android/pulse/PulseAdapter.java

```diff
@@ -117,23 +123,16 @@ public void setPulse(final int mode, Pulse pulse) {
         mMode = mode;
         mPulse.clear();
         mPulse.addAll(pulse.entrySet());
-        mPosition = new int[mPulse.size()];
+        //Only initialize if the positions are not provided.
+        //They are provided in constructor and coming from savedInstanceState of the Fragment.
+        if (mPositions == null || mPositions.length != mPulse.size()) {
```

Comment:
Should this init be down here? It feels this could be done at the beginning of the method or better yet in constructor.